### ISSUE TITLE
Clients connected to notify_push stop getting updates after suspend

### ISSUE THREAD

**Issue by alerque** [nextcloud/notify_push#53]
I have the notify_push app running in my Nextcloud instance. The instance in served by an Apache virtual host with a custom php-fpm pool running as the nextcloud user, and the websocket addresses are proxied by the same Apache server to the push service.

This arrangement works great for Desktop sync clients and they get notified of changes very quickly. Total server load has come down quite a bit and it feels snappier.

The main trouble I've found is with Laptops. If the sync client is running and they are suspended (lid closed or whatever), then resumed, the sync client continues to show a check mark and pretends that everything is up to date, but they will not sync anything that happened on the server while they were suspended nor will they get anything new that's happening afterwards. The only way to get them synced up again is to interfere: stop and start the client, pause and resume sync, or force sync now. Any of those user interferences work to get synchronization happening again, but the client won't automatically figure out that it's behind.

I assume it somehow thinks the WS is still open when the sever has decided to give up and they are just listening to dead air. The end user experience though is things get out of sync without a visible indication that a problem is brewing.

**Comment by rakurtz** [comment]
Mmh. I have installed the new version (0.1.6) and also restarted the system daemon. 

Interestingly it's very easy to reproduce the issue: If you have a vpn connection that routes all your traffic through and you activate it / deactivate it, the client stops to sync any changes. Only restarting the client solves that problem.

(Mac OS Client, newest version)

**Comment by alerque** [comment]
I just updated to 0.1.6 and did a quick check with a desktop client in exactly the same way I was testing before. I see no difference. I tried various periods in suspend mode. It seemed to connect when left offline for just a few seconds, but quickly showed the same symptoms as in the original report as soon as *any* sync events happen while the client is in suspend mode, when it does come back online it will be fooled into complacency and does not resume syncing no matter how many new events happen until the user manually forces a sync or restarts the client.

I wouldn't consider this bug fixed until it is actually fixed for real-world cases. It doesn't really matter if the server bit is capable of supporting something if the client doesn't know anything about it.

If it's a bug in the client say so and lets open a bug report on that repo, but it's a bit disconcerting that you think the server half is fixed without even knowing how the client works. For now it's easy to see how people consider this *symptomatically* a bug in this project since the client handled this case just fine until the notify_push component here is added, then it stops working.
